bpf-rust-programs/prediction-poll/program_data/src/collection.rs

```diff
@@ -0,0 +1,49 @@
+use alloc::vec::Vec;
+use solana_sdk_types::SolPubkey;
+
+pub struct CollectionData<'a> {
+    poll_count: &'a mut u32,
+    polls: &'a mut [SolPubkey],
+}
+
+impl<'a> CollectionData<'a> {
+    pub fn from_bytes(data: &'a mut [u8]) -> Self {
+        let (poll_count, polls) = data.split_at_mut(4);
+        Self {
+            poll_count: unsafe { &mut *(&mut poll_count[0] as *mut u8 as *mut u32) },
```

Comment:
Could this be: `poll_count: &mut *(data as *mut u32),`

Reply:
Not allowed :/
```
error[E0606]: casting `&'a mut [u8]` as `*mut u32` is invalid
```